### ISSUE TITLE
jpeg_codec_differentiable returns all-NaN at jpeg_quality=0, and the tests sample it at random

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4205]
### Describe the bug

`jpeg_codec_differentiable` returns an **all-NaN** image when `jpeg_quality` is exactly `0`, on every device, with a NaN gradient to match. The docstring documents the supported range as `[0, 100]` (`kornia/enhance/jpeg.py:519`), so `0` is a documented input, not a misuse.

The cause is an unguarded division in `_jpeg_quality_to_scale` (`kornia/enhance/jpeg.py:236-241`):

```python
scale = _differentiable_polynomial_floor(
    torch.where(
        compression_strength < 50,
        5000.0 / compression_strength,   # <- inf at quality 0
        200.0 - 2.0 * compression_strength,
    )
)
```

At `compression_strength == 0` the taken branch is `5000.0 / 0` -> `inf`, and the polynomial floor of `inf` propagates NaN through the whole codec.

### Reproduction steps

```python
import torch, kornia

img = torch.rand(2, 3, 32, 32)
q = torch.tensor([0.0, 50.0], requires_grad=True)
out = kornia.enhance.jpeg_codec_differentiable(img, q)
print(out.isnan().flatten(1).any(1).tolist())   # [True, False]
out.nansum().backward()
print(q.grad.tolist())                          # [nan, 0.000528...]
```

Verified identical on `cpu` and `mps`, torch 2.14.0, kornia at `1e322cb1`. It is elementwise, so only the batch entries whose quality is `0` are poisoned. Quality `1` is fine (`5000/1 = 5000`), so the defect is a single point.

### Why this matters beyond the edge case: it makes CI intermittently red

`tests/enhance/test_jpeg.py` draws the quality with **`torch.randint(low=0, high=100, ...)` at 16 sites**, unseeded, so every run has a real chance of drawing a `0` and producing NaN. Tests that only assert a shape survive it; tests that compare values do not — for example `TestDiffJPEG::test_keyword_argument` calls the codec twice with the *same* inputs and asserts closeness, which fails because `NaN != NaN`:

```
Mismatched elements: 3072 / 6144 (50.0%)
Greatest absolute difference: nan at index (1, 0, 0, 0) (up to 1e-05 allowed)
```

The 50.0% is one whole image of the `B=2` batch — the entry that drew quality `0`.

Measured rate, 20 consecutive plain CPU runs of that one file:

```
$ for i in $(seq 1 20); do pytest tests/enhance/test_jpeg.py --device=cpu --dtype=float32 -q; done
run 10: 1 failed, 11 passed
FAILED 1 / 20
```

So roughly **5% of runs of this file go red on any job**, CPU included — it is not MPS-specific. I hit it in a full-suite MPS run while re-recording the strict known-failure baseline for #4202, where an unlisted failure is a hard gate failure; it passed on the immediately following identical run.

### Expected behavior

Two things, and I think both are wanted:

1. `jpeg_quality=0` should return a finite, heavily-compressed image rather than NaN — clamp the strength away from zero before the division, or document and validate the range as `[1, 100]` and reject `0`. Whichever is chosen, the docstring and the code should agree; today they do not.
2. `tests/enhance/test_jpeg.py` should not sample the failing value at random. Drawing from `low=1` (or seeding the draws) removes the flake independently of the fix, and a dedicated pinned case for the boundary is better coverage than a 1-in-100 lottery spread over 16 tests.

Worth noting for whoever picks this up: `RandomJPEG` defaults to `jpeg_quality=50.0` and samples a continuous uniform, so hitting exactly `0.0` through the augmentation has measure zero. The integer `randint` draws in the tests are what make it reachable in practice.

### Environment

```
kornia 1e322cb1 | torch 2.14.0 | python 3.11 | OS macOS 26.5.1 arm64 | MPS=True
```
Reproduced on both the `cpu` and `mps` devices; the defect is in device-independent Python arithmetic.

Posted on behalf of @ducha-aiki by Claude (Claude Opus 5).

**Comment by modelpath-dev** [comment]
I'll work on this. Could you assign it to me?

I'll pin a regression test (or stabilize the flake) and keep the change small.

**Comment by ducha-aiki** [comment]
@modelpath-dev — yours, thanks for picking it up. GitHub will not let me set the assignee field for a non-collaborator, so this comment is the claim marker; nobody else should start on it.

One steer before you begin, because the issue offers two halves and they are not interchangeable:

**Please fix the NaN, not just the flake.** Changing `torch.randint(low=0, ...)` to `low=1` in `tests/enhance/test_jpeg.py` makes CI green and leaves `jpeg_codec_differentiable(img, quality=0)` returning an all-NaN image with a NaN gradient, at a value the docstring advertises as supported (`kornia/enhance/jpeg.py:519` documents `[0, 100]`). That would be papering over the defect with the test change that hides it. The repo's preference is to fix the wart rather than document or dodge it.

So the shape I would like:

1. **`kornia/enhance/jpeg.py:236-241`** — decide what `quality=0` means and make the code and the docstring agree. Either it returns a finite, maximally-compressed image (clamp the strength away from zero before `5000.0 / compression_strength`), or `[0, 100]` becomes `[1, 100]` and `0` is rejected with a real validation error. I lean towards the first — `0` reads as "worst quality", not "invalid" — but either is defensible as long as both sides say the same thing. If you clamp, please pick the bound by measuring rather than by taste, and say in the PR what you measured: the quantity that matters is where `5000 / q` stops being representable and where the polynomial floor stops being finite, and those are dtype-dependent. Note that the smallest positive value of a dtype is its smallest *subnormal*, not `torch.finfo(dtype).tiny` — flooring at `tiny` silently de-normalises the whole subnormal band, which we have got wrong here before.
2. **A pinned regression test** at the boundary, asserting finiteness of both the output *and* the gradient — the issue's repro shows `q.grad` is NaN too, and an output-only assertion would miss a fix that only cleans up the forward pass. Please make it fail on `main`: run it against unmodified `kornia/` first and paste that output in the PR. A test that passes on both sides pins nothing.
3. **Then** the sampling change in `tests/enhance/test_jpeg.py`. With (1) in place `low=0` would be safe, but a value drawn at random across 16 sites is poor coverage either way — a dedicated pinned case beats a 1-in-100 lottery. Whether you also raise the `low` is your call; if you leave it at `0` it becomes a genuine (if slow) fuzz over the fixed boundary.

Worth knowing while you work: this reproduces identically on `cpu`, so you do not need an accelerator to see it, and `quality=1` is fine (`5000/1`), so it really is a single point rather than a region. Run at least `pixi run test-module tests/enhance/test_jpeg.py --dtype=float32,float64` plus `pixi run pre-commit-all` before you push, and add a `CHANGELOG.md` entry under `## Unreleased` — for the docstring-vs-behaviour half, say what used to happen as well as what happens now.

Ping me here if the clamp-versus-reject decision turns out to be less obvious than it looks from the outside; it is the only real design choice in the change and I would rather settle it before you write the test than after.

Posted on behalf of @ducha-aiki by Claude (Claude Opus 5).
